content/posts/irods-development-update-march-2024.md

```diff
@@ -0,0 +1,293 @@
+Title: iRODS Development Update: March 2024
+Date: 2024-03-13 14:00
+Author: Kory Draughn
+Slug: irods-development-update-march-2024
+Status: published
+
+
+Hello reader,
+
+It's time for another development update. For the server, the trim API received an important bug fix which makes it so that it can no longer trim the last replica. The `--link` option for `iput` and `irsync` was updated to correctly ignore symlinks. Before this change, the `--link` option would result in immediate termination of the operation.
+
+There's been a huge amount of activity in regard to the build and packaging effort. We now offer packages built against libc++ (for the 4.3 series) and libstdc++ (for the upcoming 5.x series). Dependency declarations for externals packages are better. The Clang compiler has been bumped to version 13.0.1. Users experiencing packaging issues involving the MySQL database plugin and MariaDB packages will be happy to know we've resolved the problem. And finally, the CMake package version file has been updated to meet the requirements of the version numbering change. As a result, the Docker-based Development Environment has been updated to support libstdc++.
+
+Work on the Kubernetes Testing Environment is progressing. For those unaware, we've been working on new test infrastructure designed to run all tests on every GitHub pull request, hands-free. As of now, we can launch iRODS Providers and Consumers and run core tests. There's even a GUI! We see a lot of promise, but there's still a lot of work to do.
+
+Over in plugin land, we've finally released the Indexing capability plugin for iRODS 4.3.1. The plugin now uses **Boost.Beast** for HTTP communication with Elasticsearch. This release includes support for TLS communication and Basic authentication with Elasticsearch. The plugin is designed to work with Elasticsearch 7 and later. An important note about this release - the **document-type** rule engine plugin is no longer provided by the Indexing capability plugin and as a result, you'll need to remove the **document-type** rule engine plugin from your server_config.json file. See the README at [https://github.com/irods/irods_capability_indexing/tree/88bbf7ba6a8276c1349ad41afb0346fbb0ca9ac8](https://github.com/irods/irods_capability_indexing/tree/88bbf7ba6a8276c1349ad41afb0346fbb0ca9ac8) for details.
```

Comment:
"Over in plugin land, we've finally released" -> "We have released"

Reply:
Done.